extensions/amp-story/1.0/amp-story-share.js

```diff
@@ -277,13 +315,40 @@ export class ShareWidget {
     // TODO(alanorozco): Listen for proper tap event (i.e. fastclick)
     listen(linkShareButton, 'click', e => {
       e.preventDefault();
-      this.copyUrlToClipboard_();
+      this.copyUrlToClipboard_(this.root.querySelector('#page-share').checked);
     });
   }
 
-  /** @private */
-  copyUrlToClipboard_() {
-    const url = Services.documentInfoForDoc(this.getAmpDoc_()).canonicalUrl;
+  /** On desktop mode, with branching, users should be able to share a story
+   * starting from a specific page.
+   * @private */
+  maybeAddPageShareButton_() {
+    const isDesktopUi =
+      this.storeService_.get(
+          StateProperty.UI_STATE) === UIType.DESKTOP_FULLBLEED;
+
+    if (isExperimentOn(this.win, 'amp-story-branching') && isDesktopUi) {
+
+      const sharePageCheck =
+        renderAsElement(this.win.document, SHARE_PAGE_TEMPLATE);
+      // sharePageCheck.querySelector(
+      //     '#page-share-span').innerHTML = 'Share this page';
+
+      this.root.appendChild(sharePageCheck);
+    }
+  }
+
+  /**
+   * @param {boolean} opt_sharePage
+   * @private
+   */
+  copyUrlToClipboard_(opt_sharePage) {
+    const currentPageId = this.storeService_.get(StateProperty.CURRENT_PAGE_ID);
+    const url =
```

Comment:
There's some duplication here. The difference is what's appended, so you can have that be the ternary value

```js
const shouldAddFragment = 
    (isExperimentOn(this.win, 'amp-story-branching') && opt_sharePage)
const url = Services.documentInfoForDoc(this.getAmpDoc_()).canonicalUrl + 
    (shouldAddFragment ? '#page=' + currentPageId : '')
```

Reply:
This makes sense and is much more readable. Thanks, Carlos!